worker/credentialvalidator/worker.go

```diff
@@ -67,20 +63,34 @@ func NewWorker(config Config) (worker.Worker, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	w, err := config.Facade.WatchCredential(mc.CloudCredential)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
+
+	// It is possible that this model is on the cloud that does not
+	// necessarily require authentication. Consequently, a model
+	// is created without a set credential. However, later on,
+	// a credential may be added to a model. This worker will need to
+	// be restarted as there will be no credential watcher set to react to
+	// changes.
+	// TODO (anastasiamac 2018-05-30) when model-credential relationship no
+	// longer resides on model itself but has a dedicated document in mongo,
+	// we can have a watcher that will support the above corner case.
 
 	v := &validator{
 		validatorFacade: config.Facade,
 		credential:      mc,
-		watcher:         w,
 	}
 
-	err = catacomb.Invoke(catacomb.Plan{
+	plan := catacomb.Plan{
 		Site: &v.catacomb,
-		Work: v.loop,
+		Work: v.loopNoWatcher,
+	}
+
+	if mc.CloudCredential != "" {
+		var err error
+		v.watcher, err = config.Facade.WatchCredential(mc.CloudCredential)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		plan.Work = v.loop
```

Comment:
We don't need 2 loops. What you do is have a possibly nil channel. If nil, it will never provide a value. eg

var watcherChanges NotifyChannel
if cred {
   watcherChanges = watcher.Changes()
}
select {
  case _, ok := <-watcherChanges
}